aws/raw-to-ami.sh

```diff
@@ -0,0 +1,177 @@
+#!/usr/bin/env bash
+
+set -e
+[ "$DEBUG" == 'true' ] && set -x
+
+ARGC=$#
+if [ $ARGC -ne 3 ]; then
+    echo "USAGE: $(basename $0) <Image Path> <S3 Bucket Name> <Region>"
+    exit 1
+fi
+IMAGE_FILE_PATH=$1
+BUCKET_NAME=$2
+REGION=$3
+
+IMAGE_NAME="$(basename -- ${IMAGE_FILE_PATH})"
+AMI_NAME="${IMAGE_NAME%.*}"
+FORMAT="${IMAGE_NAME#*.}"
+
+TMPDIR=$(mktemp -d)
+IMAGE_IMPORT_JSON_FILE="${TMPDIR}/image-import.json"
+AMI_REGISTER_JSON_FILE="${TMPDIR}/register-ami.json"
+TRUST_POLICY_JSON_FILE="${TMPDIR}/trust-policy.json"
+ROLE_POLICY_JSON_FILE="${TMPDIR}/role-policy.json"
+BUCKET_POLICY_JSON_FILE="${TMPDIR}/bucket-policy.json"
+
+pre_checks() {
+	[[ -x "$(command -v jq)" ]] || { echo "jq is not installed" 1>&2 ; exit 1; }
+	[[ -x "$(command -v aws)" ]] || { echo "aws is not installed" 1>&2 ; exit 1; }
+	[[ "$FORMAT" == "raw"  ]] || { echo "image must be \"raw\", convert with: \"qemu-img convert ${IMAGE_NAME} ${AMI_NAME}.raw\"" 1>&2 ; exit 1; }
+	aws sts get-caller-identity &>/dev/null || { echo "awc cli missing credentials"; exit 1; }
+}
+
+create_bucket() {
+	echo "Create s3 Bucket"
+	aws s3api create-bucket  --bucket ${BUCKET_NAME} --region ${REGION}
```

Comment:
when the bucket already exists, will this return an error? will the script exit immediately because of 'set -e'? Would it be appropriate to ignore the error if it means that the bucket already exists?

Reply:
Indeed, I'll address this